File: docs/guides/php/src/saveImageClassifications.php
<?php

use Algolia\AlgoliaSearch\Api\SearchClient;

$getImageLabels = function ($imageURL, $objectID, $scoreLimit) {
    // Implement your image classification logic here
    return ['objectID' => '', 'imageURL' => '', 'objects' => []];
};

$client = SearchClient::create('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY');

$records = [];

$hits = $client->browseObjects('<YOUR_INDEX_NAME>');
foreach ($hits as $hit) {
    $imageURL = $hit['imageURL'];
    $records[] = $getImageLabels($imageURL, $hit['objectID'], 0.5);
}

$client->partialUpdateObjects(
    '<YOUR_INDEX_NAME>',
    $records,
    true,
);

File: docs/guides/ruby/saveImageClassifications.rb
require "algolia"

def get_image_labels(image_url, object_id, score_limit)
  # Implement your image classification logic here
  {"objectID" => "", "imageURL" => "", "objects" => []}
end

client = Algolia::SearchClient.create("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")

records = []

client.browse_objects(
  "<YOUR_INDEX_NAME>"
) do |resp|
  resp.hits.each do |hit|
    image_url = hit["imageURL"]
    records << get_image_labels(image_url, hit["objectID"], 0.5)
  end
end

client.partial_update_objects("<YOUR_INDEX_NAME>", records, true)
